src/neo-vm/Debugger.cs

```diff
@@ -44,10 +39,22 @@ public bool RemoveBreakPoint(byte[] script_hash, uint position)
             return true;
         }
 
+        void ExecuteNext()
+        {
+            engine.ExecuteNext();
+            if (engine.State == VMState.NONE && engine.InvocationStack.Count > 0 && break_points.Count > 0)
+            {
+                if (break_points.TryGetValue(engine.CurrentContext.ScriptHash, out HashSet<uint> hashset) && hashset.Contains((uint)engine.CurrentContext.InstructionPointer))
+                    engine.State = VMState.BREAK;
+            }
+        }
+
         public void StepInto()
         {
             if (engine.State.HasFlag(VMState.HALT) || engine.State.HasFlag(VMState.FAULT)) return;
-            engine.ExecuteNext();
+
+            ExecuteNext();
```

Comment:
I think `StepInto()` only execute one instruction, it won't skip the break points. So there is no need to check the break points.